docs/src/modules/java/pages/quickstart/cr-value-entity-kafka-java.adoc

```diff
@@ -0,0 +1,160 @@
+----
+--
+
+. Create a `customer_action.proto` file and save it in the `src/main/proto/customer/action` directory.
+
+. Add declarations for:
++
+* The protobuf syntax version, `proto3`.
+* The package name, `customer.action`.
+* The required Java outer classname, `CustomerAction`. Messages defined in this file will be generated as inner classes.
+* Import `customer/domain/customer_domain.proto` and Akka Serverless `akkaserverless/annotations.proto`.
++
+[source,proto,indent=0]
+.src/main/proto/customer/action/customer_action.proto
+----
+include::example$java-customer-registry-kafka-quickstart/src/main/proto/customer/action/customer_action.proto[tag=declarations]
+----
+
+. Add the service definition. The service definition is annotated with `akkaserverless.codegen` indicating we want to generate an Action for this service.
+. Add declarations for:
++
+* Listening to `customer.domain.CustomerState` events from the value entity `customer` by using the option `eventing.in`.
+* Publishing to the Kafka topic `customer_upserts` by using the option `eventing.out`.
++
+[source,proto,indent=0]
+.src/main/proto/customer/action/customer_action.proto
+----
+include::example$java-customer-registry-kafka-quickstart/src/main/proto/customer/action/customer_action.proto[tag=service]
+----
+
+. Run `mvn compile` from the project root directory to generate source classes in which you add business logic.
++
+----
+mvn compile
+----
+
+== Create an Action
+
+Actions are stateless functions that can be triggered in multiple ways. In this case the action is triggered by each event `customer.domain.CustomerState` received by the Value Entity `customer.domain.Customer`. 
+
+. If it's not open already, open `src/main/java/customer/action/CustomerStateSubscriptionAction.java` for editing.
+
+. Modify the `onUpsertState` method by adding the logic to handle the action. The complete method should include the following:
++
+[source, java]
+.src/main/java/customer/action/CustomerStateSubscriptionAction.java
+----
+include::example$java-customer-registry-kafka-quickstart/src/main/java/customer/action/CustomerStateSubscriptionAction.java[tag=upsert]
+----
++
+* The incoming message contains the request data from your client and this action just passes it along to the xref:java:actions-publishing-subscribing.adoc[Pub/Sub] mechanism of choice. In this example Kafka.
+
+== Package and deploy your service
+
+To compile, build the container image, and publish it to your container registry, follow these steps
+
+. From the root project directory, compile the source code using Maven:
++
+[source, command line]
+----
+mvn compile
+----
+
+. Use the `deploy` target to build the container image and publish it to your container registry. At the end of this command Maven will show you the container image URL you'll need in the next part of this quickstart.
++
+[source, command line]
+----
+mvn deploy
+----
+
+. Sign in to your Akka Serverless account at: {console}
```

Comment:
This section is no longer correct and should be removed. Deployments are only done using `akkasls` so the step would be

```
. If you do not have a project, run `akkasls projects new <project name> --region us-east1`
. To deploy to your new project, you'll have to set it as the active project,  by running: 'akkasls config set project <project name>'
. To deploy your service, run `akkasls services deploy kafka-value-entity <image URL from previous step>`
```

Reply:
I've just been updating the deploy and invoke sections for the other quickstarts in #836

Reply:
I followed same pattern as #836